pkg/testutils/dhcp4server/leasepool/lease.go

```diff
@@ -0,0 +1,102 @@
+package leasepool
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+)
+
+type LeaseStatus int
+
+const (
+	Free     LeaseStatus = 0
+	Reserved LeaseStatus = 1
+	Active   LeaseStatus = 2
+)
+
+type Lease struct {
+	IP         net.IP           // The IP of the Lease
+	Status     LeaseStatus      // Are Reserved, Active or Free
+	MACAddress net.HardwareAddr // Mac Address of the Device
+	ClientID   []byte           // ClientID of the request
+	Hostname   string           // Hostname From option 12
+	Expiry     time.Time        // Expiry Time
+}
+
+// leaseMarshal is a mirror of Lease used for marshalling, since
+// net.HardwareAddr has no native marshalling capability.
+type leaseMarshal struct {
+	IP         string
+	Status     int
+	MACAddress string
+	ClientID   string
+	Hostname   string
+	Expiry     time.Time
+}
+
+func (l Lease) MarshalJSON() ([]byte, error) {
+	return json.Marshal(leaseMarshal{
+		IP:         l.IP.String(),
+		Status:     int(l.Status),
+		MACAddress: l.MACAddress.String(),
+		ClientID:   hex.EncodeToString(l.ClientID),
+		Hostname:   l.Hostname,
+		Expiry:     l.Expiry,
+	})
+}
+
+func (l *Lease) UnmarshalJSON(data []byte) error {
+	stringUnMarshal := leaseMarshal{}
+	err := json.Unmarshal(data, &stringUnMarshal)
+	if err != nil {
+		return err
+	}
+
+	l.IP = net.ParseIP(stringUnMarshal.IP)
```

Comment:
Maybe check if the parse failed here.

Reply:
This whole file is just a copy-paste of an old dependency. It's silly code just for testing; I didn't want to fix it up.